concourse/client/api.py

```diff
@@ -221,9 +221,24 @@ def build_events(self, build_id):
     @ensure_annotations
     def trigger_resource_check(self, pipeline_name: str, resource_name: str):
         url = self.routes.resource_check(pipeline_name=pipeline_name, resource_name=resource_name)
-        # Resource checks are triggered by a POST with an empty JSON-document as body against
-        # the resource's check-url
-        self._post(url, body='{}')
+        max_retries = 5
+
+        def _try_resource_check(current_attempt):
+            # Resource checks are triggered by a POST with an empty JSON-document as body against
+            # the resource's check-url
+            response = self.request_builder.post(url, check_http_code=False, body='{}', headers={})
```

Comment:
another nit: we should use `json={}` (i know this was not there in the original code)